### ISSUE TITLE
Over 100% CPU for clojure JVM processes after waking up 

### ISSUE THREAD

**Issue by lucywang000** [nrepl/nrepl#216]
Whenever I wake up my computer from a long (3h+) sleep, the cpu usage for all clojure java processes (either launched by lein or shadow-cljs) would skyrocket for a while, but they would cool down after half a minute. The longer the sleep, the longer it takes for the cpu usage to cool down.

So I use VisualVM to get some profiling data, and it seems it’s related to nrepl:

<img width="1095" alt="Screen Shot 2020-11-05 at 8 01 21 AM" src="https://user-images.githubusercontent.com/30067222/98188571-33afc700-1f4e-11eb-8e5d-887da08e043f.png">

Does anyone has a clue?

Here are the versions:

OS: MacOS 10.15
emacs: 27.1
cider: CIDER 1.0.0snapshot (package: 20200916.1152)
cider-nrepl: 0.25.4
nrepl: 0.8.2
refactor-nrepl: 2.5.0

This happened since a long time and I have upgraded almost all these libs (even emacs), but the problem is always there.

(also posted in [clojureverse form](https://clojureverse.org/t/over-100-cpu-for-clojure-jvm-processes-after-waking-up/6790) )

**Comment by shen-tian** [comment]
Does this happen consistently, for all processes created by lein and shadow? It will be hard to pin down without reproduction steps.

To reason through this a bit:

- the inner `bencode/read-byte` is called once for each token read, to see the type of the next token. **if** this is called a lot of times, it suggests a lot of messages, or messages containing many tokens. 
- the behaviour where longer sleep = more stall time, and eventual recovery, suggests that a backlog of messages is being built up during sleep.

Try [logging](https://docs.cider.mx/cider/troubleshooting.html#debugging-the-communication-with-nrepl) nrepl messages to see if you spot anything?

**Comment by lucywang000** [comment]
Yeah it happens consistently, for all clojure repl processes (lein or shadow).

Thanks for the debug tip, I'll try that.

**Comment by bbatsov** [comment]
@lucywang000 Does this still happen with nREPL 1.3? @alexander-yakushev did a lot of internal improvements in recent releases, so the issue might have been addressed by some of them.

**Comment by lucywang000** [comment]
i no longer see this issue, so i think we can close it
